expression/aggregation/descriptor.go

```diff
@@ -172,7 +191,7 @@ func (a *AggFuncDesc) GetAggFunc() Aggregation {
 	case ast.AggFuncAvg:
 		return &avgFunction{aggFunction: aggFunc}
 	case ast.AggFuncGroupConcat:
```

Comment:
Can we fetch MaxGroupConcatLen here?
put `MaxGroupConcatLen uint64` in `AggFuncDesc` is also somewhat inappropriate

Reply:
The problem of fetching `MaxGroupConcatLen` there is that currently many places in the framework we can not get the `context.Context`, and it takes efforts to change this......

I think it's reasonable that `AggFuncDesc` contains `MaxGroupConcatLen` because `MaxGroupConcatLen` describes the attribute of the `GROUP_CONCAT` aggregation function, that's what `AggFuncDesc` expected to do.

Reply:
It seems not that difficult to change this?😂
Most of them are test functions,
we can pass MockContext into them.

``` go
executor/builder.go
949:            aggFunc := aggDesc.GetAggFunc()  // We can get context easily here.
981:            aggFunc := aggDesc.GetAggFunc()

expression/aggregation/aggregation.go
210:            newAggFunc = desc.GetAggFunc() // ditto
215:            newAggFunc = desc.GetAggFunc()
228:    return desc.GetAggFunc()

expression/aggregation/aggregation_test.go
57:     avgFunc := NewAggFuncDesc(s.ctx, ast.AggFuncAvg, []expression.Expression{col}, false).GetAggFunc()
75:     distinctAvgFunc := NewAggFuncDesc(s.ctx, ast.AggFuncAvg, []expression.Expression{col}, true).GetAggFunc()
105:    avgFunc := aggFunc.GetAggFunc()
122:    sumFunc := NewAggFuncDesc(s.ctx, ast.AggFuncSum, []expression.Expression{col}, false).GetAggFunc()
142:    distinctSumFunc := NewAggFuncDesc(s.ctx, ast.AggFuncSum, []expression.Expression{col}, true).GetAggFunc()
158:    bitAndFunc := NewAggFuncDesc(s.ctx, ast.AggFuncBitAnd, []expression.Expression{col}, false).GetAggFunc()
233:    bitOrFunc := NewAggFuncDesc(s.ctx, ast.AggFuncBitOr, []expression.Expression{col}, false).GetAggFunc()
316:    bitXorFunc := NewAggFuncDesc(s.ctx, ast.AggFuncBitXor, []expression.Expression{col}, false).GetAggFunc()
391:    countFunc := NewAggFuncDesc(s.ctx, ast.AggFuncCount, []expression.Expression{col}, false).GetAggFunc()
410:    distinctCountFunc := NewAggFuncDesc(s.ctx, ast.AggFuncCount, []expression.Expression{col}, true).GetAggFunc()
430:    concatFunc := NewAggFuncDesc(s.ctx, ast.AggFuncGroupConcat, []expression.Expression{col, sep}, false).GetAggFunc()
456:    distinctConcatFunc := NewAggFuncDesc(s.ctx, ast.AggFuncGroupConcat, []expression.Expression{col, sep}, true).GetAggFunc()
478:    firstRowFunc := NewAggFuncDesc(s.ctx, ast.AggFuncFirstRow, []expression.Expression{col}, false).GetAggFunc()
502:    maxFunc := NewAggFuncDesc(s.ctx, ast.AggFuncMax, []expression.Expression{col}, false).GetAggFunc()
503:    minFunc := NewAggFuncDesc(s.ctx, ast.AggFuncMin, []expression.Expression{col}, false).GetAggFunc()

expression/aggregation/bench_test.go
32:     fun := NewAggFuncDesc(ctx, ast.AggFuncAvg, []expression.Expression{col}, false).GetAggFunc()
46:     fun := NewAggFuncDesc(ctx, ast.AggFuncAvg, []expression.Expression{col}, false).GetAggFunc()
61:     fun := NewAggFuncDesc(ctx, ast.AggFuncAvg, []expression.Expression{col}, true).GetAggFunc()
75:     fun := NewAggFuncDesc(ctx, ast.AggFuncAvg, []expression.Expression{col}, true).GetAggFunc()

```

Reply:
```go
expression/aggregation/aggregation.go
210:            newAggFunc = desc.GetAggFunc() // ditto
215:            newAggFunc = desc.GetAggFunc()
228:    return desc.GetAggFunc()
```
It seems that we can't get `ctx` directly here,  more need to be changed.   =。=   
I'll try here. :)